app/server/skychat/commands/impl/core/AccountPlugin.ts

```diff
@@ -59,6 +73,33 @@ export class AccountPlugin extends Plugin {
         connection.send('message', message.sanitized());
     }
 
+    async handleChangeUsername(param: string, connection: Connection): Promise<void> {
+
+        const user = connection.session.user;
+        const username = param.split(' ')[0];
+        const password = param.split(' ').slice(1).join(' ');
+
+        if (! user.testHashedPassword(UserController.hashPassword(user.id, user.username, password))) {
+            throw new Error('Invalid password');
+        }
+
+        if (await UserController.getUserByUsername(username)) {
+            throw new Error('This username already exists');
+        }
+
+        if (user.money < AccountPlugin.CHANGE_USERNAME_PRICE) {
+            throw new Error(`You need at least ${AccountPlugin.CHANGE_USERNAME_PRICE} to change username`)
+        }
+
+        await UserController.buy(user, AccountPlugin.CHANGE_USERNAME_PRICE);
+
+        const message = UserController.createNeutralMessage(`Your username has been changed to ${username}`);
+        connection.send('message', message.sanitized());
+
+        await UserController.changeUsername(user, username, password);
+        connection.session.identifier = user.username.toLowerCase();
+    }
+
```

Comment:
If the user is OP, the `.env.json` should be updated to also rename this username to the new one.
Otherwise, someone else could become op by also changing its username.